framework/meta/src/cmd/standalone/upgrade/upgrade_selector.rs

```diff
@@ -69,41 +78,32 @@ fn upgrade_function_selector(dir: &RelevantDirectory) {
         print_upgrading(dir);
     }
 
-    match dir.upgrade_in_progress {
-        Some((_, "0.31.0")) => {
-            upgrade_to_31_0(dir);
-        },
-        Some((_, "0.32.0")) => {
-            upgrade_to_32_0(dir);
-        },
-        Some((_, "0.39.0")) => {
-            upgrade_to_39_0(dir);
-        },
-        Some((_, "0.45.0")) => {
-            upgrade_to_45_0(dir);
-        },
-        Some((from_version, to_version)) => {
-            version_bump_in_cargo_toml(&dir.path, from_version, to_version);
-        },
-        None => {},
+    if let Some((from_version, to_version)) = dir.upgrade_in_progress {
+        match to_version.version.to_string().as_str() {
+            "0.31.0" => upgrade_to_31_0(dir),
+            "0.32.0" => upgrade_to_32_0(dir),
+            "0.39.0" => upgrade_to_39_0(dir),
+            "0.45.0" => upgrade_to_45_0(dir),
+            _ => version_bump_in_cargo_toml(&dir.path, from_version, to_version),
+        }
     }
 }
 
 fn upgrade_post_processing(dir: &RelevantDirectory, settings: &UpgradeSettings) {
-    match dir.upgrade_in_progress {
-        Some((_, "0.28.0")) | Some((_, "0.29.0")) | Some((_, "0.30.0")) | Some((_, "0.31.0"))
-        | Some((_, "0.32.0")) | Some((_, "0.33.0")) | Some((_, "0.34.0")) | Some((_, "0.35.0"))
-        | Some((_, "0.36.0")) | Some((_, "0.37.0")) | Some((_, "0.40.0")) | Some((_, "0.41.0"))
-        | Some((_, "0.42.0")) | Some((_, "0.43.0")) | Some((_, "0.44.0")) | Some((_, "0.45.2"))
-        | Some((_, "0.46.0")) => {
+    if let Some((_, to_version)) = dir.upgrade_in_progress {
+        if [
+            "0.28.0", "0.29.0", "0.30.0", "0.31.0", "0.32.0", "0.33.0", "0.34.0", "0.35.0",
```

Comment:
We should extract this to a const array, same as the big version list, using the same macro. Can also reside in version_history.rs. You can call it something like `CHECK_AFTER_UPGRADE_TO`